src/controls/dynamicForm/DynamicForm.tsx

```diff
@@ -597,17 +597,17 @@ export class DynamicForm extends React.Component<
             hiddenName = response.value;
             termSetId = field.TermSetId;
             anchorId = field.AnchorId;
-            if (item && item[field.InternalName] && item[field.InternalName].results) {
-              item[field.InternalName].results.forEach((element) => {
+            if (item && item[field.InternalName] && item[field.InternalName]) {
+              item[field.InternalName].forEach((element) => {
                 selectedTags.push({
                   key: element.TermGuid,
                   name: element.Label,
                 });
               });
 
               defaultValue = selectedTags;
-            } else if (defaultValue && defaultValue.results) {
-              defaultValue.results.forEach((element) => {
+            } else if (defaultValue) {
```

Comment:
Lines 609-616 can also be removed (as default values are handled below), they raised an error when loading a new item with default terms configured on the field.